imports/plugins/included/product-variant/client/containers/productGridItemsContainer.js

```diff
@@ -0,0 +1,277 @@
+          // Reset sessions ver of selected products
+          Session.set("productGrid/selectedProducts", []);
+
+          // Reset the action view of selected products
+          Reaction.setActionView({
+            label: "Grid Settings",
+            i18nKeyLabel: "gridSettingsPanel.title",
+            template: "productSettings",
+            type: "product",
+            data: {}
+          });
+        }
+      }
+    }
+  }
+
+  productPath = () => {
+    if (this.props.product) {
+      let handle = this.props.product.handle;
+
+      if (this.props.product.__published) {
+        handle = this.props.product.__published.handle;
+      }
+
+      return Reaction.Router.pathFor("product", {
+        hash: {
+          handle
+        }
+      });
+    }
+
+    return "/";
+  }
+
+  positions = () => {
+    const tag = ReactionProduct.getTag();
+    return this.props.product.positions && this.props.product.positions[tag] || {};
+  }
+
+  weightClass = () => {
+    const positions = this.positions();
+    const weight = positions.weight || 0;
+    switch (weight) {
+      case 1:
+        return "product-medium";
+      case 2:
+        return "product-large";
+      default:
+        return "product-small";
+    }
+  }
+
+  isSelected= () => {
+    if (Reaction.isPreview() === false) {
+      return _.includes(Session.get("productGrid/selectedProducts"), this.props.product._id) ? "active" : "";
+    }
+    return false;
+  }
+
+  productMedia = () => {
+    const media = Media.findOne({
+      "metadata.productId": this.props.product._id,
+      "metadata.toGrid": 1
+    }, {
+      sort: { "metadata.priority": 1, "uploadedAt": 1 }
+    });
+
+    return media instanceof FS.File ? media : false;
+  }
+
+  additionalProductMedia = () => {
+    const mediaArray = Media.find({
+      "metadata.productId": this.props.product._id,
+      "metadata.priority": {
+        $gt: 0
+      },
+      "metadata.toGrid": 1
+    }, { limit: 3 });
+
+    return mediaArray.count() > 1 ? mediaArray : false;
+  }
+
+  isMediumWeight = () => {
+    const positions = this.positions();
+    const weight = positions.weight || 0;
+
+    return weight === 1;
+  }
+
+  displayPrice = () => {
+    if (this.props.product.price && this.props.product.price.range) {
+      return this.props.product.price.range;
+    }
+  }
+
+  onDoubleClick = () => {
+    const product = this.props.product;
+    const handle = product.__published && product.__published.handle || product.handle;
+
+    Reaction.Router.go("product", {
+      handle: handle
+    });
+
+    Reaction.setActionView({
+      i18nKeyLabel: "productDetailEdit.productSettings",
+      label: "Product Settings",
+      template: "ProductAdmin"
+    });
+  }
+
+  onClick = (event) => {
+    const product = this.props.product;
+
+    if (Reaction.hasPermission("createProduct") && Reaction.isPreview() === false) {
+      event.preventDefault();
```

Comment:
rather than having this duplicated in both the `if` and `else` branches can we just put `event.preventDefault()` at the top?